src/shell/grabs/moving.rs

```diff
@@ -200,6 +223,69 @@ impl MoveGrabState {
 struct NotSend<T>(pub T);
 unsafe impl<T> Send for NotSend<T> {}
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SnappingZone {
+    TopMaxim,
+    TopSnap,
+    Left,
+    Right,
+    Bottom,
+}
+
+const SNAPPING_RANGE: i32 = 12;
+
+impl SnappingZone {
+    pub fn relative_geometry(
+        &self,
+        output_geometry: Rectangle<i32, Logical>,
+    ) -> Rectangle<i32, Local> {
+        let (loc, size) = match self {
+            SnappingZone::TopMaxim => (
+                output_geometry.loc,
+                Size::from((output_geometry.size.w, SNAPPING_RANGE / 2)),
+            ),
+            SnappingZone::TopSnap => (
+                Point::from((
+                    output_geometry.loc.x,
+                    output_geometry.loc.y + SNAPPING_RANGE / 2,
+                )),
+                Size::from((output_geometry.size.w, SNAPPING_RANGE / 2)),
+            ),
+            SnappingZone::Left => (
+                output_geometry.loc,
+                Size::from((SNAPPING_RANGE, output_geometry.size.h)),
+            ),
+            SnappingZone::Right => (
+                Point::from((
+                    output_geometry.loc.x + output_geometry.size.w - SNAPPING_RANGE,
+                    output_geometry.loc.y,
+                )),
+                Size::from((SNAPPING_RANGE, output_geometry.size.h)),
+            ),
+            SnappingZone::Bottom => (
+                Point::from((
+                    output_geometry.loc.x,
+                    output_geometry.loc.y + output_geometry.size.h - SNAPPING_RANGE,
+                )),
+                Size::from((output_geometry.size.w, SNAPPING_RANGE)),
+            ),
+        };
+        Rectangle::from_loc_and_size(loc, size).as_local()
```

Comment:
I think it would probably be good, if this took panels into account. Meaning imo the zone should be all of the `exclusive_zone` at the specific border + SNAPPING_RANGE.

So dragging onto a panel would still activate the zone, but there would always be an additional range next to every dock/panel/etc.

Reply:
fixed in bee0b664cc83aec19f5da5100570c8f17af3c987